yaml_edit/lib/src/editor.dart

```diff
@@ -438,7 +425,7 @@ class YamlEditor {
     final parentNode = _traverse(collectionPath);
 
     if (parentNode is YamlList) {
-      edit = removeInList(this, parentNode, keyOrIndex);
+      edit = removeInList(this, parentNode, keyOrIndex as int);
```

Comment:
We should probably also check that `keyOrIndex` is actually a valid index and throw `PathError` if not from here too..

Reply:
I'll be adding tests that specifically covers index problems with lists and found on my local copy that this is already covered because `remove()` uses `_traverse()`.